apps/allocations/app/components/Panel/NewBudget.js

```diff
@@ -139,29 +148,53 @@ class NewBudget extends React.Component {
                 onChange={this.handleSelectToken}
               />
             )}
-
+            <Duration />
           </InputGroup>
         </Field>
       </Form>
     )
   }
 }
 
-const NewBudgetWrap = props => {
-  const { appState: { tokens = [] } } = useAragonApi()
+const Duration = () => {
+  const { appState: { period } } = useAragonApi()
   const theme = useTheme()
-  const saveBudget = useSaveBudget()
+  //Set duration to default, then calculate if 'period' is set
+  let duration = DEFAULT_DURATION
+  if(period) {
+    const { startDate, endDate } = period
+    duration = endDate - startDate + 1000
+  }
+
+  const days = Math.round(duration/MS_PER_DAY)
+  let string
+  if(days%7 === 0) {
+    const weeks = days/7
+    string = weeks === 1 ? 'week' : `${weeks} weeks`
+  } else {
+    string = days === 1 ? 'day' : `${days} days`
+  }
 
   return (
-    <NewBudget
-      saveBudget={saveBudget}
-      tokens={tokens}
-      theme={theme}
-      {...props}
-    />
+    <Text
+      css={`
+        white-space: nowrap;
+        line-height: 40px;
+        padding-left: ${2 * GU}px;
+        font-size: 16px;
+        color: ${theme.contentSecondary};
+      `}
+    >
+      every {string}
+    </Text>
   )
 }
 
+const NewBudgetWrap = props => {
+  const { appState: { tokens = [] } } = useAragonApi()
+  return <NewBudget tokens={tokens} {...props} />
```

Comment:
Why stop passing in `saveBudget` and `theme`? This seems like a bug.

Reply:
I think that `saveBudget` must have been removed when I rebased. But I believe it gets passed in `{...props}', no?

`theme` was only being used by a `Text` component that has been moved to `Duration` so `NewBudget` doesn't need it anymore

Reply:
`saveBudget` was being defined in `NewBudgetWrap` and passed into `NewBudget`. I think you'll find that saving budgets no longer works.